validator/sawtooth_validator/journal/chain.py

```diff
@@ -43,161 +54,246 @@ def __init__(self, consensus,
         self._executor = executor
         self._squash_handler = squash_handler
 
-    @property
-    def new_block(self):
-        return self._new_block
+        self._result = {
+            'new_block': new_block,
+            'chain_head': chain_head,
+            'new_chain': [],
+            'cur_chain': [],
+            'committed_batches': [],
+            'uncommitted_batches': [],
+        }
 
-    @property
-    def chain_head(self):
-        return self._chain_head
+    def _is_block_complete(self, blkw):
```

Comment:
1) The completer is not the only source of blocks. the blocks can be loaded from the BlockCache or the BlockStore. Both of risk of having corrupted data. Long term it is not clear the policy of the BlockCache will be only to contain completed blocks. 
2) During testing blocks do not go thru the completer. Best to fail early on bad data.